classes/Cart.php

```diff
@@ -5113,4 +5061,107 @@ public function getTaxCountry(): Country
 
         return new Country($taxAddress->id_country);
     }
+
+    /**
+     * Alter raw cart details to adapt to display use case.
+     *
+     * @param array $summary
+     * @param bool $refresh
+     *
+     * @return array
+     */
+    private function alterSummaryForDisplay(array $summary, bool $refresh = false): array
+    {
+        $context = Context::getContext();
+        $currency = new Currency($this->id_currency);
+
+        $products = $this->getProducts($refresh);
```

Comment:
```suggestion
        $products = $summary['products'];
```
No need to make two queries (especially when `refresh` is true)